utils/swift_build_support/swift_build_support/products/swift_testing.py

```diff
@@ -0,0 +1,203 @@
+# swift_build_support/products/swift_testing.py -----------------*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2024 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+
+import os
+
+from . import cmake_product
+from . import product
+from . import swift
+
+
+class SwiftTesting(product.Product):
+    @classmethod
+    def is_build_script_impl_product(cls):
+        return False
+
+    @classmethod
+    def is_before_build_script_impl_product(cls):
+        return False
+
+    @classmethod
+    def product_source_name(cls):
+        return "swift-testing"
+
+    @classmethod
+    def get_dependencies(cls):
+        return [swift.Swift]
+
+    def should_build(self, host_target):
+        return True
+
+    def should_test(self, host_target):
+        return False
+
+    def should_install(self, host_target):
+        return self.args.install_swift_testing_macros
+
+    def _impl_product(self, host_target):
+        build_root = os.path.dirname(self.build_dir)
+        build_dir = os.path.join(
+            build_root, '%s-%s' % (self.product_name(), host_target))
+
+        return SwiftTestingImpl(
+            args=self.args,
+            toolchain=self.toolchain,
+            source_dir=self.source_dir,
+            build_dir=build_dir)
+
+    def _build_impl(self, host_target):
+        self._impl_product(host_target).build(host_target)
+
+    def build(self, host_target):
+        self._build_impl(host_target)
+
+        # For Darwin host, 'build' is only called for the builder.
+        # Manually iterate tor the cross compile hosts.
+        if self.has_cross_compile_hosts() and self.is_darwin_host(host_target):
+            for target in self.args.cross_compile_hosts:
+                self._build_impl(target)
+
+        # FIXME: build testing library for 'stdlib_deployment_targets'?
+        pass
+
+    def _install_impl(self, host_target):
+        self._impl_product(host_target).install(host_target)
+
+    def install(self, host_target):
+        self._install_impl(host_target)
+
+        # For Darwin host, 'install' is only called for the builder.
+        # Manually iterate tor the cross compile hosts.
+        if self.has_cross_compile_hosts() and self.is_darwin_host(host_target):
+            for target in self.args.cross_compile_hosts:
+                self._install_impl(target)
+
+
+class SwiftTestingImpl(cmake_product.CMakeProduct):
+    def build(self, host_target):
+        override_version = None
+        if host_target.startswith('macosx'):
+            override_version = '10.15'
```

Comment:
We may want to override this only if the macOS deployment version is lower than the desired one -- this way if the default changes or the user uses a newer one we are not stuck with an old version

This suggestion is an expedient way to achieve this in this context --  we should likely prefer the functions defined in https://github.com/swiftlang/swift/blob/main/utils/build_swift/build_swift/versions.py

```suggestion
            def version_parse(version_string):
                return list(map(lambda x: int(x), version_string.split(".")))

            if version_parse(self.args.darwin_deployment_version_osx) < (10, 15):
                 override_version = '10.15'
```

Reply:
Attempting a stab using the `Version` class:

```suggestion
            from build_swift.build_swift.versions import Version

            if Version(self.args.darwin_deployment_version_osx) < Version('10.15'):
                 override_version = '10.15'
```